pkg/gardener/tasks/awsmachineimages.go

```diff
@@ -60,6 +60,18 @@ func collectAWSMachineImages(ctx context.Context, payload CollectCPMachineImages
 	logger := asynqutils.GetLogger(ctx)
 	logger.Info("collecting machine images", "cloud_profile", payload.CloudProfileName)
 	items := make([]models.CloudProfileAWSImage, 0)
+
+	// represents the complex key of the item. used for
+	// deduplicating records.
+	type key struct {
```

Comment:
Instead of introducing an intermediate struct for the only purpose of detecting duplicates perhaps a better approach would be something along the lines:

``` go
...

compareFunc := func(a, b ...) {
   // Compare the items
}

// Sort and compact
slices.SortFunc(items, compareFunc)
slices.Compact(items)

// Persist in db
```

Or do it using `slices.CompactFunc` instead.